jetty-server/src/test/java/org/eclipse/jetty/server/RequestTest.java

```diff
@@ -1965,6 +1966,7 @@ public void testPushBuilder()
         request.getResponse().getHttpFields().add(new HttpCookie.SetCookieHttpField(new HttpCookie("good", "thumbsup", 100), CookieCompliance.RFC6265));
         request.getResponse().getHttpFields().add(new HttpCookie.SetCookieHttpField(new HttpCookie("bonza", "bewdy", 1), CookieCompliance.RFC6265));
         request.getResponse().getHttpFields().add(new HttpCookie.SetCookieHttpField(new HttpCookie("bad", "thumbsdown", 0), CookieCompliance.RFC6265));
+        request.getResponse().getHttpFields().add(new HttpField(HttpHeader.SET_COOKIE, new HttpCookie("ugly", "thumbsdown", 100).getSetCookie(CookieCompliance.RFC6265)));
```

Comment:
We should get the example cookie that the OP is setting and use it here.
There might be an aspect of RFC6265 we've missed.

Reply:
This usage, of having a `HttpField` with an `HttpCookie` object as a value isn't parsing like you think.
Add a Same-Site to that cookie, and see if the resulting cookie from the Push has that Same-Site attribute.